server/src/main/java/org/apache/uniffle/server/DefaultFlushEventHandler.java

```diff
@@ -0,0 +1,135 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.config.RssBaseConf;
+import org.apache.uniffle.common.exception.RssException;
+import org.apache.uniffle.common.util.ThreadUtils;
+import org.apache.uniffle.server.storage.StorageManager;
+import org.apache.uniffle.storage.common.HdfsStorage;
+import org.apache.uniffle.storage.common.LocalStorage;
+import org.apache.uniffle.storage.common.Storage;
+import org.apache.uniffle.storage.util.StorageType;
+
+public class DefaultFlushEventHandler implements FlushEventHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultFlushEventHandler.class);
+
+  private final ShuffleServerConf shuffleServerConf;
+  private final StorageManager storageManager;
+  private Executor localFileThreadPoolExecutor;
+  private Executor hdfsThreadPoolExecutor;
+  private final StorageType storageType;
+  protected final BlockingQueue<ShuffleDataFlushEvent> flushQueue = Queues.newLinkedBlockingQueue();
+  private Consumer<ShuffleDataFlushEvent> eventConsumer;
+
+  public DefaultFlushEventHandler(ShuffleServerConf conf, StorageManager storageManager,
+      Consumer<ShuffleDataFlushEvent> eventConsumer) {
+    this.shuffleServerConf = conf;
+    this.storageType = StorageType.valueOf(shuffleServerConf.get(RssBaseConf.RSS_STORAGE_TYPE));
+    this.storageManager = storageManager;
+    this.eventConsumer = eventConsumer;
+    initFlushEventExecutor();
+  }
+
+  @Override
+  public void handle(ShuffleDataFlushEvent event) {
+    if (!flushQueue.offer(event)) {
+      LOG.warn("Flush queue is full, discard event: " + event);
+    } else {
+      ShuffleServerMetrics.gaugeEventQueueSize.inc();
+    }
+  }
+
+  private void handleEventAndUpdateMetrics(ShuffleDataFlushEvent event, boolean isLocalFile) {
+    try {
+      eventConsumer.accept(event);
+    } finally {
+      if (isLocalFile) {
+        ShuffleServerMetrics.counterLocalFileEventFlush.inc();
+      } else {
+        ShuffleServerMetrics.counterHdfsEventFlush.inc();
+      }
+    }
+  }
+
+  protected void initFlushEventExecutor() {
+    if (StorageType.withLocalfile(storageType)) {
+      int poolSize = shuffleServerConf.getInteger(ShuffleServerConf.SERVER_FLUSH_LOCALFILE_THREAD_POOL_SIZE);
+      localFileThreadPoolExecutor = createFlushEventExecutor(poolSize, "LocalFileFlushEventThreadPool");
+    }
+    if (StorageType.withHDFS(storageType)) {
+      int poolSize = shuffleServerConf.getInteger(ShuffleServerConf.SERVER_FLUSH_HDFS_THREAD_POOL_SIZE);
+      hdfsThreadPoolExecutor = createFlushEventExecutor(poolSize, "HdfsFlushEventThreadPool");
+    }
+    startEventProcessor();
+  }
+
+  private void startEventProcessor() {
+    // the thread for flush data
+    Thread processEventThread = new Thread(this::eventLoop);
+    processEventThread.setName("ProcessEventThread");
+    processEventThread.setDaemon(true);
+    processEventThread.start();
+  }
+
+  protected void eventLoop() {
+    while (true) {
```

Comment:
Could we have a variable to control whether stop the eventloop? Could we add the method `stop`?